src/neo/SmartContract/ApplicationEngine.Runtime.cs

```diff
@@ -295,6 +295,26 @@ protected internal void RuntimeNotify(byte[] eventName, Array state)
             using MemoryStream ms = new(MaxNotificationSize);
             using BinaryWriter writer = new(ms, Utility.StrictUTF8, true);
             BinarySerializer.Serialize(writer, state, MaxNotificationSize);
+            Queue<StackItem> queue = new();
```

Comment:
Why are we putting these into a queue in order to validate them? Can't we just build a new recursive method that calls itself to check items in array or map recurively?

Reply:
Stack overflows if we use recursive method.